jwql/website/apps/jwql/data_containers.py

```diff
@@ -316,22 +447,20 @@ def get_current_flagged_anomalies(rootname, instrument):
 
```

Comment:
Could you add docstrings for instrument and n_match? Is n_match always 1 since a True value from the database will translate into a 1?

Reply:
Ah I see, from the call in get_anomaly_forms(), n_match will be the number of rootnames. I thought the call to first() in line 457 would return only the first matching row?

Reply:
Good catch -- I'll update the docstring.

N_match is for distinguishing which anomalies belong to a whole exposure group and which belong only to a single file in a group.  If you're only checking for a single file, n_match should be 1.  If you're checking for an exposure group, n_match should be the total number of files in the group.  This is needed as an argument because not every rootfile has an entry in the anomaly table: they only appear there if someone has marked an anomaly at some point.

The first() in line 457 takes the first entry for each rootfile because it follows the groupby function on the dataframe.  This keeps the last marked anomaly state for each rootfile, in case it has multiple entries in the database.

For example, if there are 2 files in an exposure group and one is marked with 'persistence' and 'crosstalk' and the other is marked with 'crosstalk' only, then the current anomalies for the group will include only 'crosstalk'.

Reply:
Yes, that's right.  The rootname is used as a starts-with filter, so if jw86600008001_02101_00001_guider2 is passed, only that file will be matched.  If jw86600008001_02101_00001 is passed, then either jw86600008001_02101_00001_guider1 or jw86600008001_02101_00001_guider2 will be matched.  I'll clarify the example root name in the docstring.  

The group root name is passed only in the context of the exposure pages: the image pages still pass the full file root name.